base/pkg/dir.jl

```diff
@@ -54,8 +55,15 @@ function init(meta::AbstractString=DEFAULT_META, branch::AbstractString=META_BRA
                 close(io)
             end
         end
+        #Move TEMP to METADATA
+        mkpath(dir)
+        Base.mv(joinpath(temp_dir,"METADATA"), metadata_dir)
+        Base.mv(joinpath(temp_dir,"REQUIRE"), joinpath(dir,"REQUIRE"))
+        Base.mv(joinpath(temp_dir,"META_BRANCH"), joinpath(dir,"META_BRANCH"))
+        rm(temp_dir)
```

Comment:
@jakebolewski If the directory is not empty it then will error.
I had it the other way initially but thought that if some other change comes around which isn't accounted for in this then it will error.

I am not too sure whether this is a good thing though. It means if somebody adds an extra item to `Pkg.init()` it needs to be explicitly handled whereas if I had moved `temp_dir` to `dir` instead of each item individually that wouldn't have been a problem.